dokka-subprojects/analysis-kotlin-api/src/test/kotlin/org/jetbrains/dokka/analysis/test/jvm/java/JavadocAnalysisTest.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.dokka.analysis.test.jvm.java
+
+import org.jetbrains.dokka.analysis.test.api.javaTestProject
+import org.jetbrains.dokka.analysis.test.api.parse
+import org.jetbrains.dokka.model.SourceSetDependent
+import org.jetbrains.dokka.model.doc.*
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class JavadocAnalysisTest {
```

Comment:
Could you please add a single test for `@param Bar` on a class/function (not `<Bar>`, but `Bar`)? To verify that it either resolves `Bar` correctly, or it doesn't resolve it at all.

It's highlighted in red by IDEA, but Dokka 1.8.20 resolves it correctly in the resulting documentation. Whether it's right by specification and whether Dokka should support it is a different question, I'm not proposing to change the resolution logic now, I want the test to only "freeze" this behaviour so that we are notified in case it changes. And once/if it does change - we'll figure out what to do with it

Reply:
Ok, I will add.
But, as far as I remember, during implementation, I found that both java and kotlin analysis for `param` tag doesn't validate if such parameter exists. So f.e., if you will have class like this (kotlin):
```kotlin
/**
 * @param non_existent documentation
 **/
fun something(param: String) {}
```
In the end, documentation for `non_existent` param will be present in generated HTML file. Same for Java.
Also, there will be no links for params - as they have nowhere to go - as parameters are already shown on the same page under this documentation

You can take a look here on what I mean: https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html (parameters are underlined but have no links)